pkg/scheduler/internal/cache/cache.go

```diff
@@ -506,17 +506,19 @@ func (cache *cacheImpl) AddPod(pod *v1.Pod) error {
 	currState, ok := cache.podStates[key]
 	switch {
 	case ok && cache.assumedPods.Has(key):
+		// When assuming, we've already added the Pod to cache,
+		// Just update here to make sure the Pod's status is up-to-date.
+		if err = cache.updatePod(currState.pod, pod); err != nil {
```

Comment:
Revisit the code, I think we should call updatePod after L506. If pod exists in `podStates` but not in `assumedPods`, we should also update the pod, right?

Reply:
I don't think so. This `AddPod` func should be called only once after Pod is scheduled, and the case "pod exists in podStates but not in assumedPods" shouldn't be happened.
If that case happens, that means `AddPod` func is somehow called twice (or more) for the same Pod. And, I think we don't need to do anything and just returning an error in L528 is enough.

Reply:
>  and the case "pod exists in podStates but not in assumedPods" shouldn't be happened. 

Actually it's possible, a pod can be added without scheduler like specifying the nodeName in podSpec. But I think you're right. We'll not add the same pod twice.

Reply:
> a pod can be added without scheduler like specifying the nodeName in podSpec. 

Ahh, like an unscheduled Pod is created -> a user (or a controller) specify nodeName before the scheduler schedule it -- that's a rare case, but can happen.
Let me change here as you suggest, thanks

Reply:
But this matches the `case !ok`, no?

Reply:
Fixed

Reply:
Sorry, commented at the same time (again)

> But this matches the case !ok, no?

Ah, right. nvm, I'll revert the commit I just added.